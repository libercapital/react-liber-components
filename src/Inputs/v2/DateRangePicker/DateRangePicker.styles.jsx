
import React from 'react';
import styled, { css } from 'styled-components';
import { MdCalendarBlank, MdCloseCircle } from '../../../Icons';
import { handleThemeFromObject } from '../../../Themes';

const getTheme = (value, fallback) => css`
    ${props => handleThemeFromObject(props.theme, value, fallback)}
`;

export const Container = styled.div`
  position: relative;
`;

const CleanIcon = ({ error, hide, ...props }) => <MdCalendarBlank {...props} />;  // eslint-disable-line
export const Icon = styled(CleanIcon)`
  --size: 20px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
  fill: ${props => (props.error
    ? getTheme('colors.error', '#dc3545')
    : getTheme('colors.dark-text70', '#798e9b'))};
  position: absolute;
  right: 16px;
  top: calc(50% - 10px);
  pointer-events: none;
  opacity: ${({ hide }) => (hide ? '0' : '1')};
  transition: opacity 250ms cubic-bezier(0.23,1,0.32,1) 0ms;


`;

const closeShow = css`
  opacity: 1;
  pointer-events: all;
`;

const ClearClose = ({ show, ...props }) => <MdCloseCircle {...props} />;  // eslint-disable-line
export const Close = styled(ClearClose)`
  --size: 20px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
  fill: ${getTheme('colors.dark-text70', '#798e9b')};
  position: absolute;
  right: 16px;
  top: calc(50% - 10px);
  opacity: 0;
  pointer-events: none;
  ${({ show }) => show && closeShow}
  transition: opacity 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
  z-index: 2;
  cursor: pointer;
`;
