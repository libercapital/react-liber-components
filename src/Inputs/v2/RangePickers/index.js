export { default as Input } from './Input';
export { default as Range } from './Range';
export { InputShell, Label, Hint, Error, Input as StyledInput } from './Range.styles';
