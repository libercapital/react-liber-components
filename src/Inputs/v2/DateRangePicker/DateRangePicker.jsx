import React, { PureComponent as Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Container, Icon, Close } from './DateRangePicker.styles';

import { SimpleDropDown } from '../../../DropDown';
import { ClickOutside } from '../../../Util';
import { Input, Range as RangeSelector, InputShell, Label, Hint, Error } from '../RangePickers';

const DATE_FORMAT = 'DD/MM/YYYY';

class DateRangePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      clear: false,
      startDate: '',
      endDate: '',
      displayDate: props.defaultDisplayDate || moment(),
    };
    this.input = React.createRef();
  }

  focusInput = () => {
    this.input.current.focus();
  };

  handleInputFocus = () => {
    this.setState({ open: true });
  };

  handleOutsideClick = () => {
    this.setState({ open: false });
  };

  setCallbackValue = (start = '', end = '') => {
    if (start === '' && end === '') {
      return '';
    }
    if (end === '') {
      return start.trim();
    }
    return `${start.trim()} a ${end.trim()}`;
  };

  getParsedValue = () => {
    const { value } = this.props;
    if (value) {
      let [startDate = '', endDate = ''] = value.split('a');
      startDate = startDate.trim();
      endDate = endDate.trim();
      return { startDate, endDate };
    }
    return {};
  };

  timeout = delay => {
    this.delay = setTimeout(() => {
      this.setState({ open: false });
    }, delay);
  };

  handleDateSelection = (mStartDate, mEndDate) => {
    const { onChange, closeDelay, closeOnSelect } = this.props;
    const startDate = mStartDate ? mStartDate.format('DD/MM/YYYY') : '';
    const endDate = mEndDate ? mEndDate.format('DD/MM/YYYY') : '';
    if (onChange) {
      onChange(this.setCallbackValue(startDate, endDate));
    } else {
      const displayDate = mStartDate ? moment(mStartDate).startOf('month') : undefined;
      this.setState({ startDate, endDate, displayDate });
    }
    if (moment(mEndDate).isValid() && closeOnSelect) {
      this.timeout(closeDelay);
    }
  };

  handleStartChange = endDate => event => {
    const { onChange } = this.props;
    if (onChange) {
      onChange(this.setCallbackValue(event.target.value, endDate));
    } else {
      const displayDate = moment(event.target.value, DATE_FORMAT).startOf('month');
      this.setState({
        startDate: event.target.value,
        displayDate: displayDate.isValid() ? displayDate : undefined,
      });
    }
  };

  handleEndChange = startDate => event => {
    const { onChange, closeDelay, closeOnSelect } = this.props;
    const { value } = event.target;
    if (onChange) {
      onChange(this.setCallbackValue(startDate, value));
    } else {
      this.setState({
        endDate: value,
      });
    }
    if (value.length >= 10 && moment(value, DATE_FORMAT).isValid() && closeOnSelect) {
      this.timeout(closeDelay);
    }
  };

  handleOnMouseEnter = () => {
    const { startDate: stateStartDate, endDate: stateEndDate } = this.state;
    const { startDate = stateStartDate, endDate = stateEndDate } = this.getParsedValue();
    if (startDate.trim() !== '' || endDate.trim() !== '') {
      this.setState({ clear: true });
    }
  };

  handleOnMouseLeave = () => {
    this.setState({ clear: false });
  };

  handleNavigate = displayDate => {
    this.setState({ displayDate });
  };

  clearInput = () => {
    const { onChange } = this.props;
    if (onChange) {
      onChange('');
      this.setState({ clear: false });
    } else {
      this.setState({
        startDate: '',
        endDate: '',
        clear: false,
      });
    }
  };

  renderShell = (startDate, endDate, open, error, disabled) => (
    <InputShell
      onClick={this.focusInput}
      onMouseEnter={this.handleOnMouseEnter}
      handleOnMouseLeave={this.handleOnMouseLeave}
      open={open}
      error={error}
      disabled={disabled}
    >
      <Input
        ref={this.input}
        onFocus={this.handleInputFocus}
        fields={[
          {
            value: startDate,
            mask: '99/99/9999',
            separator: 'a',
            onChange: !disabled ? this.handleStartChange(endDate) : undefined,
            id: 'start_day_field',
            size: '87px',
            disabled,
          },
          {
            value: endDate,
            mask: '99/99/9999',
            onChange: !disabled ? this.handleEndChange(startDate) : undefined,
            id: 'end_day_field',
            size: '87px',
            disabled,
          },
        ]}
      />
    </InputShell>
  );

  render() {
    const {
      open,
      clear,
      startDate: stateStartDate,
      endDate: stateEndDate,
      displayDate: stateDisplayDate,
    } = this.state;
    const {
      label,
      openLabel,
      disabled,
      errorMessage,
      placeholder,
      hintText,
      displayDate = stateDisplayDate,
      onNavigate = this.handleNavigate,
      position,
      autoPosition,
    } = this.props;
    const { startDate = stateStartDate, endDate = stateEndDate } = this.getParsedValue();
    return (
      <ClickOutside onOutsideClick={this.handleOutsideClick}>
        <Container>
          {label !== '' && (
            <Label
              open={(openLabel || open || startDate !== '' || endDate !== '') && !disabled}
              error={errorMessage !== ''}
              disabled={disabled}
            >
              {label}
            </Label>
          )}

          <Hint visible={(open || !label) && !(startDate !== '' || endDate !== '')}>
            {placeholder || hintText}
          </Hint>
          {errorMessage !== '' && <Error visible={errorMessage !== ''}>{errorMessage}</Error>}
          <SimpleDropDown
            open={open && !disabled}
            position={position}
            autoPosition={autoPosition}
            customButton={this.renderShell(startDate, endDate, open, errorMessage !== '', disabled)}
            hasOutsideLabel={label !== ''}
          >
            <RangeSelector
              showTime={false}
              startDate={startDate.length >= 10 ? moment(startDate, DATE_FORMAT) : null}
              endDate={endDate.length >= 10 ? moment(endDate, DATE_FORMAT) : null}
              onChangeDates={!disabled ? this.handleDateSelection : undefined}
              displayDate={displayDate}
              onNavigate={onNavigate}
            />
          </SimpleDropDown>
          <Icon hide={clear} error={errorMessage !== ''} />
          <Close show={clear} onMouseEnter={this.handleOnMouseEnter} onClick={this.clearInput} />
        </Container>
      </ClickOutside>
    );
  }
}

DateRangePicker.propTypes = {
  label: PropTypes.string,
  openLabel: PropTypes.bool,
  value: PropTypes.string,
  disabled: PropTypes.bool,
  errorMessage: PropTypes.string,
  placeholder: PropTypes.string,
  hintText: PropTypes.string,
  displayDate: PropTypes.instanceOf(moment),
  defaultDisplayDate: PropTypes.instanceOf(moment),
  onChange: PropTypes.func,
  onNavigate: PropTypes.func,
  closeOnSelect: PropTypes.bool,
  closeDelay: PropTypes.number,
  position: PropTypes.string,
  autoPosition: PropTypes.bool,
};

DateRangePicker.defaultProps = {
  label: '',
  openLabel: false,
  value: null,
  disabled: false,
  errorMessage: '',
  placeholder: '',
  hintText: '',
  displayDate: undefined,
  defaultDisplayDate: undefined,
  onChange: undefined,
  onNavigate: undefined,
  closeOnSelect: true,
  closeDelay: 500,
  position: 'bottom-left',
  autoPosition: true,
};

export default DateRangePicker;
