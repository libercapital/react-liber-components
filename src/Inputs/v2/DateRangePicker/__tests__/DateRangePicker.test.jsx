import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import moment from 'moment';
import DateRangePicker from '../DateRangePicker';
import { SimpleDropDown } from '../../../../DropDown';
import { Input, Range as RangeSelector, StyledInput } from '../../RangePickers';

describe('Test for DateRangePicker', () => {
  let clock;
  configure({ adapter: new Adapter() });
  beforeEach(() => {
    const now = new Date('2019-06-07');
    clock = sinon.useFakeTimers(now.getTime());
  });
  afterEach(() => {
    clock.restore();
  });
  it('should render correctly without props', () => {
    const wrapper = shallow(<DateRangePicker />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with label', () => {
    const wrapper = shallow(<DateRangePicker label="label" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with label and openLabel', () => {
    const wrapper = shallow(<DateRangePicker label="label" openLabel />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with value', () => {
    const wrapper = shallow(
      <DateRangePicker value="15/05/2019 a 16/05/2019" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly disabled', () => {
    const wrapper = shallow(<DateRangePicker disabled />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with erro Message', () => {
    const wrapper = shallow(<DateRangePicker errorMessage="Error Message" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with hintText', () => {
    const wrapper = shallow(<DateRangePicker hintText="Hint Message" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with placeholder', () => {
    const wrapper = shallow(<DateRangePicker placeholder="placeholder" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with displayDate', () => {
    const wrapper = shallow(
      <DateRangePicker displayDate={moment('2019-07-07', 'YYYY-MM-DD')} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with defaultDisplayDate', () => {
    const wrapper = shallow(
      <DateRangePicker
        defaultDisplayDate={moment('2019-07-07', 'YYYY-MM-DD')}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should select dates on RangeSelector changes', () => {
    const wrapper = shallow(<DateRangePicker />);
    const selector = wrapper.find(RangeSelector);
    const start = moment('2019-06-07', 'YYYY-MM-DD');
    const end = moment('2019-07-07', 'YYYY-MM-DD');
    selector.simulate('changeDates', start, end);
    expect(wrapper.state('startDate')).toEqual(start.format('DD/MM/YYYY'));
    expect(wrapper.state('endDate')).toEqual(end.format('DD/MM/YYYY'));
    expect(wrapper.state('displayDate')).toEqual(
      moment(start).startOf('month'),
    );
  });
  it('should select startDate state on startDate input Change', () => {
    const wrapper = shallow(<DateRangePicker />);
    const dd = wrapper.find(SimpleDropDown).dive();
    const inputContainer = dd.find(Input).dive();
    const inputs = inputContainer.find(StyledInput);
    inputs
      .at(0)
      .dive()
      .simulate('change', { target: { value: '07/06/2019' } });
    expect(wrapper.state('startDate')).toEqual('07/06/2019');
    expect(wrapper.state('displayDate')).toEqual(
      moment('07/06/2019', 'DD/MM/YYYY').startOf('month'),
    );
  });
  it('should select endDate state on endDate input Change', () => {
    const wrapper = shallow(<DateRangePicker />);
    const dd = wrapper.find(SimpleDropDown).dive();
    const inputContainer = dd.find(Input).dive();
    const inputs = inputContainer.find(StyledInput);
    inputs
      .at(1)
      .dive()
      .simulate('change', { target: { value: '07/07/2019' } });
    expect(wrapper.state('endDate')).toEqual('07/07/2019');
  });
  it('should call onChange on RangeSelector changes when controlled', () => {
    const onChange = sinon.spy();
    const wrapper = shallow(<DateRangePicker onChange={onChange} />);
    const selector = wrapper.find(RangeSelector);
    const start = moment('2019-06-07', 'YYYY-MM-DD');
    const end = moment('2019-07-07', 'YYYY-MM-DD');
    selector.simulate('changeDates', start, end);
    expect(onChange.calledWith('07/06/2019 a 07/07/2019'));
  });
  it('should call onChange on startDate input Change when controlled', () => {
    const onChange = sinon.spy();
    const wrapper = shallow(<DateRangePicker onChange={onChange} />);
    const dd = wrapper.find(SimpleDropDown).dive();
    const inputContainer = dd.find(Input).dive();
    const inputs = inputContainer.find(StyledInput);
    inputs
      .at(0)
      .dive()
      .simulate('change', { target: { value: '07/06/2019' } });
    expect(onChange.calledWith('07/06/2019'));
  });
  it('should call onChange on endDate input Change when controlled', () => {
    const onChange = sinon.spy();
    const wrapper = shallow(
      <DateRangePicker onChange={onChange} value="07/05/2019" />,
    );
    const dd = wrapper.find(SimpleDropDown).dive();
    const inputContainer = dd.find(Input).dive();
    const inputs = inputContainer.find(StyledInput);
    inputs
      .at(1)
      .dive()
      .simulate('change', { target: { value: '07/06/2019' } });
    expect(onChange.calledWith('07/05/2019 a 07/06/2019'));
  });
});
