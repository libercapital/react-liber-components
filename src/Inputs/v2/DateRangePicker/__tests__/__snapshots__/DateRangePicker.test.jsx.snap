// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for DateRangePicker should render correctly disabled 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={true}
    />
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={true}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": true,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": undefined,
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": true,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": undefined,
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={null}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with defaultDisplayDate 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={true}
    />
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-07-07T03:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with displayDate 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={true}
    />
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-07-07T03:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with erro Message 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={true}
    />
    <Rangestyles__Error
      visible={true}
    >
      Error Message
    </Rangestyles__Error>
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={true}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={true}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with hintText 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={true}
    >
      Hint Message
    </Rangestyles__Hint>
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with label 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Label
      disabled={false}
      error={false}
      open={false}
    >
      label
    </Rangestyles__Label>
    <Rangestyles__Hint
      visible={false}
    />
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={true}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with label and openLabel 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Label
      disabled={false}
      error={false}
      open={true}
    >
      label
    </Rangestyles__Label>
    <Rangestyles__Hint
      visible={false}
    />
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={true}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with placeholder 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={true}
    >
      placeholder
    </Rangestyles__Hint>
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly with value 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={false}
    />
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "15/05/2019",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "16/05/2019",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={"2019-05-16T03:00:00.000Z"}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={"2019-05-15T03:00:00.000Z"}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;

exports[`Test for DateRangePicker should render correctly without props 1`] = `
<ClickOutside
  onOutsideClick={[Function]}
>
  <DateRangePickerstyles__Container>
    <Rangestyles__Hint
      visible={true}
    />
    <SimpleDropDown
      autoPosition={true}
      containerSize={false}
      customButton={
        <ForwardRef(Rangestyles__InputShell)
          disabled={false}
          error={false}
          handleOnMouseLeave={[Function]}
          onClick={[Function]}
          onMouseEnter={[Function]}
          open={false}
        >
          <Input
            fields={
              Array [
                Object {
                  "disabled": false,
                  "id": "start_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "separator": "a",
                  "size": "87px",
                  "value": "",
                },
                Object {
                  "disabled": false,
                  "id": "end_day_field",
                  "mask": "99/99/9999",
                  "onChange": [Function],
                  "size": "87px",
                  "value": "",
                },
              ]
            }
            hide={false}
            onFocus={[Function]}
          />
        </ForwardRef(Rangestyles__InputShell)>
      }
      hasOutsideLabel={false}
      label=""
      open={false}
      position="bottom-left"
      window={[Window]}
    >
      <Range
        displayDate={"2019-06-07T00:00:00.000Z"}
        endDate={null}
        onChangeDates={[Function]}
        onNavigate={[Function]}
        showTime={false}
        startDate={null}
      />
    </SimpleDropDown>
    <DateRangePickerstyles__Icon
      error={false}
      hide={false}
    />
    <DateRangePickerstyles__Close
      onClick={[Function]}
      onMouseEnter={[Function]}
      show={false}
    />
  </DateRangePickerstyles__Container>
</ClickOutside>
`;
